onnxruntime/contrib_ops/cpu/quadric/quadric_custom_op.cc

```diff
@@ -0,0 +1,274 @@
+          // return OrtValue for allocated tensor
+          ort_value = value;
+          allocated = true;
+        } else {
+          // put the allocated value into fetches so the copy logic in utils::ExecuteGraphImpl can use it
+          fetches[i] = value;
+        }
+
+        return Status::OK();
+      };
+    }
+  }
+
+  status = utils::ExecuteSubgraph(session_state_, ffm, feeds, fetches, fetch_allocators,
+                                  ExecutionMode::ORT_SEQUENTIAL, context_.GetTerminateFlag(),
+                                  context_.Logger(), context_.GetComputeStream());
+
+  ORT_RETURN_IF_ERROR(status);
+
+  return status;
+}
+
+QuadricCustomOp::QuadricCustomOp(const OpKernelInfo& info) : IControlFlowKernel(info) {
+  ONNX_NAMESPACE::GraphProto proto;
+  ORT_ENFORCE(info.GetAttr<ONNX_NAMESPACE::GraphProto>("sub_graph", &proto).IsOK());
+  ORT_IGNORE_RETURN_VALUE(proto);
+}
+
+Status QuadricCustomOp::Compute(OpKernelContext* ctx) const {
+  auto ctx_internal = static_cast<OpKernelContextInternal*>(ctx);
+  auto* session_state = ctx_internal->SubgraphSessionState("sub_graph");
+  ORT_ENFORCE(session_state, "Subgraph SessionState was not found for sub_graph attribute.");
+
+  QuadricCustomOpImpl impl{*ctx_internal, *session_state, *info_};
+  auto status = impl.Initialize();
+  ORT_RETURN_IF_ERROR(status);
+
+  status = impl.Execute(*feeds_fetches_manager_);
+
+  return Status::OK();
+}
+
+Status QuadricCustomOp::SetupSubgraphExecutionInfo(const SessionState& session_state, const std::string& attribute_name, const SessionState& subgraph_session_state) {
+  const auto& node = Node();
+  info_ = std::make_unique<QuadricCustomOp::Info>(node, subgraph_session_state.GetGraphViewer());
+
+  const auto& subgraph_map = subgraph_session_state.GetOrtValueNameIdxMap();
+
+  std::vector<std::string> feed_names;
+  feed_names.reserve(info_->num_inputs);
+
+  const auto& input_defs = node.InputDefs();
+  for (size_t i = 0, end = info_->num_inputs; i < end; ++i) {
+    const auto* input = input_defs[i];
+    int idx;
+    ORT_ENFORCE(subgraph_map.GetIdx(input->Name(), idx).IsOK(), "Couldn't find ", input->Name(), " in subgraph_map.");
+    feed_names.push_back(input->Name());
+  }
+
+  std::unique_ptr<FeedsFetchesManager> ffm;
+  ORT_RETURN_IF_ERROR(FeedsFetchesManager::Create(feed_names, info_->subgraph_output_names, subgraph_map, ffm));
+  ORT_RETURN_IF_ERROR(utils::InitializeFeedFetchCopyInfo(subgraph_session_state, *ffm));
+
+  // find the location all the feeds will be coming from
+  std::vector<OrtDevice> feed_locations;
+  feed_locations.resize(feed_names.size());
+  for (size_t i = 0, end = feed_names.size(); i < end; ++i) {
+    const auto& location = utils::FindDeviceForValue(session_state, feed_names[i]);
+    feed_locations[i] = location;
+  }
+
+  std::vector<const OrtDevice*> fetch_locations;
+  fetch_locations.reserve(info_->num_outputs);
+
+  // we need the allocator info for each output from the If node
```

Comment:
Replace `If node`

Reply:
Done, in all locations that I could see.